### PR TITLE
Implementation: SSH Key and Port Option

### PR DESCRIPTION
**Pull request by mschnitzer**
Supersedes #1883

Difference to the old PR:
* remote_system.rb and its spec file got refactored. I added getters for some member variables of RemoteSystem because the full access doesn't make any sense.
* I implemented a function to better build the command line arguments for `ssh`, `scp`, and `rsync`
* Renamed the options to `--ssh-port` and `--ssh-identity-file`

### CODE REVIEW

File: spec/unit/remote_system_spec.rb [hunk 1]
```diff
@@ -18,7 +18,15 @@
 require_relative "spec_helper"
 
 describe RemoteSystem do
-  let(:remote_system) { RemoteSystem.new("remotehost") }
+  let(:rsys) { RemoteSystem.new("remotehost") }
```
Comment (mauromorales) [inline spec/unit/remote_system_spec.rb]: all these initializations of `RemoteSystem` would be better handled by using the subject and then putting the different arguments in variables that describe what they are ment to be. This way you could always use the subject and depending on the describe/context block define those argument variables

File: lib/remote_system.rb [hunk 2]
```diff
@@ -175,4 +184,42 @@ def remove_file(file)
...
+
+    if @ssh_identity_file
+      command_options.push("-i")
+      command_options.push(@ssh_identity_file)
+    end
+
+    command_options
+  end
+
+  def build_basic_ssh_command
```
Comment (mauromorales) [inline lib/remote_system.rb]: these two commands are so much alike that we could handle the differences by passing `:ssh` or `:scp` as an argument

Comment (mauromorales) [inline lib/remote_system.rb]: and because the logic of ssh and scp is inside the get_cmd_options maybe it actually belongs there

Comment (mschnitzer) [inline lib/remote_system.rb]: Yes, I agree here.

Comment (mauromorales) [inline lib/remote_system.rb]: weather the methods get merged or not they sound to me like they belong more to the private side of the API

File: spec/unit/remote_system_spec.rb (hunk 1 above)
Comment (mschnitzer) [inline spec/unit/remote_system_spec.rb]: Would you mind, making an example? :-) I have no idea what you mean by that.

Comment (mauromorales) [inline spec/unit/remote_system_spec.rb]: subject { RemoteSystem.new("remotehost", options) }

so you can always use subject and you only define options in the context that you need them like

let(:options_empty) { {} }
let(:options_ssh_port} { {ssh_port: 9999 } }

Comment (aduffeck) [inline spec/unit/remote_system_spec.rb]: Wouldn't all the overriden option hashes have to be called `options` to make that work?

Comment (mauromorales) [inline spec/unit/remote_system_spec.rb]: yup sorry fixing it now

**Comment by mschnitzer** [comment]
Superseded by #1898

**Pull request closed without merging**